src/systemcmds/hardfault_log/hardfault_log.c

```diff
@@ -581,6 +582,199 @@ static int write_user_stack(int fdin, int fdout, info_s *pi, char *buffer,
 	return ret;
 }
 
+/**
+ * Append hardfault data to the stored ULog file (stored in BBSRAM).
+ * @param caller
+ * @param fdin file descriptor for plain-text hardhault log to read from
+ * @return 0 on success, -errno otherwise
+ */
+static int hardfault_append_to_ulog(const char *caller, int fdin)
+{
+
+	int ret = 0;
+	int write_size_remaining = lseek(fdin, 0, SEEK_END);
+
+	if (write_size_remaining < 0) {
+		return -EINVAL;
+	}
+
+	lseek(fdin, 0, SEEK_SET);
+
+	// get the last ulog file
+	char ulog_file_name[HARDFAULT_MAX_ULOG_FILE_LEN];
+	int fd = open(HARDFAULT_ULOG_PATH, O_RDONLY);
+
+	if (fd < 0) {
+		return -errno;
+	}
+
+	if (read(fd, ulog_file_name, HARDFAULT_MAX_ULOG_FILE_LEN) != HARDFAULT_MAX_ULOG_FILE_LEN) {
+		return -errno;
+	}
+
+	close(fd);
+	ulog_file_name[HARDFAULT_MAX_ULOG_FILE_LEN - 1] = 0; //ensure null-termination
+
+	identify(caller);
+	syslog(LOG_INFO, "Appending to ULog %s\n", ulog_file_name);
+
+	// get the ulog file size
+	struct stat st;
+
+	if (stat(ulog_file_name, &st) == -1) {
+		return -errno;
+	}
+
+	const off_t ulog_file_size = st.st_size;
+
+	// open the ulog file
+	int ulog_fd = open(ulog_file_name, O_RDWR);
```

Comment:
Based on what condition exactly? I basically agree to have all ULog stuff in one place, but I'm also a bit hesitant here because it will add another dependency from hardfault to logger. And the logger module is not the defining place for ULog, the [ULog file spec](https://dev.px4.io/en/log/ulog_file_format.html) stands on its own.